quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusCompletions.java

```diff
@@ -155,8 +156,19 @@ private static void collectPropertyKeySuggestions(int offset, Node node, Propert
 			profile = key.getProfile();
 		}
 
+		Set<String> existingProperties = getExistingProperties(model);
+
 		// Completion on Quarkus properties
 		for (ExtendedConfigDescriptionBuildItem property : projectInfo.getProperties()) {
+			String propertyName = property.getPropertyName();
+
+			if (existingProperties.contains(propertyName) &&
+				node.getNodeType() == NodeType.PROPERTY_KEY &&

```

Comment:
Is `if (existingProperties.contains(propertyName))` is not enough? Why writing this extra condition?

Reply:
It is there to allow the completion shown in the gif above. When the cursor is on or right after `quarkus.http.port`, you can see that the completion options include `quarkus.http.port`. This functionality was in this test case: https://github.com/redhat-developer/quarkus-ls/blob/0ececca93a54aa020876c33c5e4196cbfd335248/quarkus.ls/com.redhat.quarkus.ls/src/test/java/com/redhat/quarkus/services/ApplicationPropertiesCompletionTest.java#L47-L49

Reply:
Thanks for your explanation, forget my comment